### PR TITLE
CUDOS-548: Refactor documentation

### PR DESCRIPTION
**Pull request by aemil145**
https://cudo-ventures.atlassian.net/browse/CUDOS-548

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,334 +1,298 @@
...
+## Help and version
 
-1. Install `WSL` [[click here]](https://docs.microsoft.com/en-us/windows/wsl/install-manual#downloading-distros)
-2. Install `npm` [[click here]](https://www.npmjs.com/get-npm)
-3. Install ` Node.js` [ [ click here ]](https://nodejs.org/en/download/package-manager/)
-4. Change `npm` default directory [ [ click here ] ](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally#manually-change-npms-default-directory)
-   - This is to avoid any permission issues with `WSL`
-5. Open `WSL`, clone the repo and cd to its directory
-6. Run both:
+Run `--help` on any `blast` command to show all available subcommands, parameters and additional information.
```
Comment (maptuhec) [inline README.md]: can we add that `help` can be used even without command like in the example below

File: README.md [hunk 2]
```diff
@@ -1,334 +1,298 @@
...
 
 ```bash
-npm install
+blast --help
 ```
+
+Run `--version` on any `blast` command to show `cudos-blast` version number.
```
Comment (maptuhec) [inline README.md]: do `blast compile --version` works?

File: README.md [hunk 3]
```diff
@@ -1,334 +1,298 @@
...
-todo
-
----
+```bash
+blast test
+```
 
-## Full commands info
+Integration tests have to be in the `{project_root}/integration_tests/` folder.  
+> Integration tests functionality is not working correctly. Only for demo purpose!
```
Comment (maptuhec) [inline README.md]: can we use something different than not working. Like the functionality is under development, sometjing like that

File: README.md [hunk 4]
```diff
@@ -1,334 +1,298 @@
...
----
+You can check the status of a non-local Cudos node by setting its URL in `blast.config.js` under `endpoint:`.
 
-### `blast unittest`
+---
+## Deploying smart contracts, interacting with them and running custom script files
 
-> Runs the unit tests of the smart contracts. 
-> The command have to be called from the project root (or the main Cargo.toml file directory).
+You can use supplied `{project_root}/scripts/deploy.js` to deploy smart contracts one by one. You can interact with them by using supplied `interact.js` as a template. You are free to use custom `.js` scripts.
```
Comment (maptuhec) [inline README.md]: should we add a code snippet with a deploy script explanation ?

File: README.md [hunk 5]
```diff
@@ -1,334 +1,298 @@
...
-- arguments: `none`
-- options: `--quiet (-q)` `optional` `Hide test log messages`
+```bash
+blast run scripts/deploy.js
+blast run scripts/interact.js
+blast run scripts/myCustomScript.js
+```
 
-**Example:**
+You can specify your own script file path. A custom network can be set with `--network` or `(-n)`. Also, signer account can be specified with `--account` or `(-a)`. Default values can be changed in `blast.config.js` under `network` or `defaultAccount`.
```
Comment (maptuhec) [inline README.md]: should we add a list of available networks?

File: README.md [hunk 6]
```diff
@@ -1,334 +1,298 @@
...
+```
 
-**Example:**
+You can specify your own script file path. A custom network can be set with `--network` or `(-n)`. Also, signer account can be specified with `--account` or `(-a)`. Default values can be changed in `blast.config.js` under `network` or `defaultAccount`. 
 
 ```bash
-blast unittest
+blast run newFolder/anotherScripts/myCustomScript.js
+blast run scripts/deploy.js -a account2
+blast run scripts/deploy.js -n cudos -a account2
```
Comment (maptuhec) [inline README.md]: we should specify what part of the account we need to provide when using -a command (name, address, pub key, etc)

**Review by maptuhec — CHANGES_REQUESTED** [review]
I would also love to see some code snippets how to write unit tests for the smart contracts, how to use the utils that are exported and etc

File: README.md (hunk 2 above)
Comment (aemil145) [reply README.md]: actually it does, --version is global and this is how it works. I can make it less confusing by not stating that --version can be called from anywhere - just `blast --version`

**Review by maptuhec — CHANGES_REQUESTED** [review]
Not all comments are addressed

File: README.md [hunk 7]
```diff
@@ -59,9 +59,11 @@ Run `--help` on any `blast` command to show all available subcommands, parameter
 
 ```bash
 blast --help
+blast node --help
+blast node start --help
```
Comment (maptuhec) [inline README.md]: will this show any different `help` information than the `blast node --help` or the ` or onyl the `blast help`

Comment (aemil145) [reply README.md]: `blast node --help` will show help for all available subcommands and `blast node start --help` will show the same description for `start` and also any optional parameters for `start` with description

File: README.md [hunk 8]
```diff
@@ -1,334 +1,350 @@
...
 ---
+## Deploying smart contracts, interacting with them and running custom script files
 
-## Full commands info
+You can use the supplied `{project_root}/scripts/deploy.js` to deploy a sample smart contract.
 
-### `blast init`
+```bash
+async function main () {
+  // functions such as 'getSigners' and 'getContractFactory' are available in global context
```
Comment (aemil145) [inline README.md:214]: If you think it would be better, I can make a list of available functions in global context

Comment (maptuhec) [reply README.md:214]: that it is a good idea, with some comments

**Review by maptuhec — CHANGES_REQUESTED** [review]
Still missing some code snippets related to unit tests

**Review by maptuhec — APPROVED** [review]

**Pull request merged**